Add DSP_LIBRARY_PATH mappings for target machines

quic-vkatoch opened the pull request:
Add DSP_LIBRARY_PATH entries in a new YAML file for the supported machines to enable consistent, machine-specific DSP library resolution.
The machine entries are sorted alphabetically by machine name to ensure maintainability and stability as new properties are added in the future. Also update the test script (check.py) to validate that paths in config.txt are consistent with entries in 00-hexagon-dsp-binaries.yaml, and update the Makefile to install the YAML file to the conf.d directory alongside the DSP binaries.

quic-vkatoch committed:
Add DSP_LIBRARY_PATH mappings for target machines

Add DSP_LIBRARY_PATH entries in a new YAML file for the
supported machines to enable consistent, machine-specific
DSP library resolution.

The machine entries are sorted alphabetically by machine name
to ensure maintainability and stability as new properties are
added in the future.

Update the Makefile to install the YAML file to the conf.d
directory alongside the DSP binaries.

Signed-off-by: Vinayak Katoch <vkatoch@qti.qualcomm.com>

lumag commented on machine_dsp_paths.yml:
```diff
@@ -0,0 +1,19 @@
+  "Thundercomm Dragonboard 845c":
+    DSP_LIBRARY_PATH: "/sdm845/Thundercomm/db845c/"
+  "Qualcomm Technologies, Inc. Robotics RB5":
+    DSP_LIBRARY_PATH: "sm8250/Thundercomm/RB5/"
```
This line is not consistent with the rest of them. Why do we need the leading and trailing slash characters?

quic-vkatoch replied:
Ack

lumag commented on machine_dsp_paths.yml:
```diff
@@ -0,0 +1,19 @@
+  "Qualcomm Technologies, Inc. QCS8300 Ride":
+    DSP_LIBRARY_PATH: "/qcs8300/Qualcomm/QCS8300-RIDE/"
+  "Qualcomm SA8775P Ride":
+    DSP_LIBRARY_PATH: "/sa8775p/Qualcomm/SA8775P-RIDE/"
```
Missing EoL

quic-vkatoch replied:
Ack

lumag commented on machine_dsp_paths.yml:
```diff
@@ -0,0 +1,19 @@
+    DSP_LIBRARY_PATH: "/qrb4210/Thundercomm/RB2/"
+  "Qualcomm Technologies, Inc. Robotics RB3gen2":
+    DSP_LIBRARY_PATH: "/qcm6490/Thundercomm/RB3gen2/"
+  "Qualcomm QCS9100 Ride Rev3":
```
You missed IQ boards which are also supported by the release.

quic-vkatoch replied:
Ack, updated.

lumag commented:
This can very quickly become unmanageable. Could you please sort them (e.g. by the DSP path) and add a comment describing sorting order? Also please extend testing scripts. New DSP subdirs must have an entry in the config file.

ekanshibu commented:
Can you also add `make install` rules for this file?

quic-vkatoch commented:
> This can very quickly become unmanageable. Could you please sort them (e.g. by the DSP path) and add a comment describing sorting order? Also please extend testing scripts. New DSP subdirs must have an entry in the config file.

Ack, I’ll sort them based on machine names so that other properties can be added if needed. I’ll also update the test scripts.

quic-vkatoch commented:
> Can you also add `make install` rules for this file?

Sure, will update.

ekanshibu commented on scripts/check.py:
```diff
@@ -2,7 +2,7 @@
 # Copyright (c) 2024-2025 Linaro Ltd.
 
-import os, re, sys
+import os, re, sys, yaml
```
can you bring this as a separate commit

quic-vkatoch replied:
Updated

quic-vkatoch committed:
scripts: Validate config.txt against 00-hexagon-dsp-binaries.yaml

Add validation to ensure paths in config.txt exist in
00-hexagon-dsp-binaries.yaml, reporting each missing path once.

Signed-off-by: Vinayak Katoch <vkatoch@qti.qualcomm.com>

lumag commented on Makefile:
```diff
@@ -10,6 +10,7 @@ clean:
 
 install:
 	./scripts/install.sh config.txt ${DESTDIR}/${DSPDIR}
+	install -D -m 0644 machine_dsp_paths.yml ${DESTDIR}/${DSPDIR}/conf.d/machine_dsp_paths.yml
```
`00-hexagon-dsp-binaries.yaml`

quic-vkatoch replied:
Ack, renamed file to 00-hexagon-dsp-binaries.yaml and updated Makefile and check.py accordingly.
